### PR TITLE
Message extraction fails when using both Globalize.formatMessage and <FormatMessage>

### PR DESCRIPTION
**Pull request by alunny**
PR is just a failing test case - I can try to fix if you don't have time to have a look.

### CODE REVIEW

**Comment by alunny** [comment]
I think I've found what's up, should have a real patch shortly.

**Comment by rxaviers** [comment]
:clap: :+1:

**Comment by alunny** [comment]
So part of the problem (at least in a test case) is that `react-globalize-compiler` doesn't transform `_globalize2['default']` into `Globalize`, as `globalize-compiler` does.

Should `react-globalize-compiler` depend on `globalize-compiler`? I don't really want to just copy that transform into here, although that would be the quickest fix.

**Comment by rxaviers** [comment]
Using the transform in here as well sounds good to me. Ideally, Globalize should adopts the same strategy for default messages that react-globalize has. Then, we'll be able to move all this stuff into globalize-compiler (where it should live).

PS: About variable names, the ideal solution should keep a table of variables (per scope) dynamically generated from `require("globalize")`, or `define(["globalize"...) calls, or`import` declaration. But, that is much more complex.

File: lib/extract-default-messages-visitors/global-globalize-format-message.js [hunk 1]
```diff
@@ -5,6 +5,8 @@ var sanitizePath = require("../util/sanitize-message-path");
 var Syntax = esprima.Syntax;
 
 module.exports = {
+  displayName: "Globalize.formatMessage visitor",
```
Comment (alunny) [inline lib/extract-default-messages-visitors/global-globalize-format-message.js:8]: I found this useful during debugging - not strictly necessary.

Comment (rxaviers) [inline lib/extract-default-messages-visitors/global-globalize-format-message.js:8]: Interesting :+1:

**Comment by alunny** [comment]
I think we'll want to do the same for `messageFormatter` calls as well as `formatMessage` btw.

**Comment by rxaviers** [comment]
Sounds good to me, thanks.

File: lib/extract-transforms/babel-globalize2-default-into-globalize.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+var esprima = require( "esprima" );
```
Comment (alunny) [inline lib/extract-transforms/babel-globalize2-default-into-globalize.js:1]: this is a duplicate file now (this was already in `extract-default-message-transforms` but not exported)

**Comment by alunny** [comment]
Should be ready to merge now.

File: lib/extract-default-messages-transforms/index.js [hunk 3]
```diff
@@ -1,4 +1,5 @@
 module.exports = [
+  require("./babel-globalize2-default-into-globalize"),
```
Comment (rxaviers) [inline lib/extract-default-messages-transforms/index.js]: For consistency, let's move `lib/extract-default-messages-transforms/babel-globalize2-default-into-globalize.js` to `lib/extract-transforms/` given it's used by both default-messages and normal compiler?

**Comment by rxaviers** [comment]
Other than one comment above, it looks good to me.

**Comment by alunny** [comment]
updated now :)

**Comment by rxaviers** [comment]
Thank you @alunny, landed and published by 0.0.6.

**Pull request closed without merging**